# Issue 22335: PyInt_AS_LONG is gone in Python 3

archive/issues_022098.json:
```json
{
    "assignees": [],
    "body": "<div id=\"comment:0\"></div>\n\nThe C++ code in `triangulations.cc` (which does not involve Cython) calls `PyInt_AS_LONG` which no longer exists in Python 3:\n\n```\n[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc: In function 'triangulations* init_triangulations(int, int, int, bool, PyObject*, PyObject*)':\n[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc:121:41: error: 'PyInt_AS_LONG' was not declared in this scope\n```\n\nLuckily, Cython defines `PyInt_AS_LONG` as an alias of `PyLong_AS_LONG` in Python 3. So there should be no compilation issues for Cython code.\n\nI include a branch to allow others to reproduce:\n\n```\nexport SAGE_PYTHON3=yes\nmake build\n```\n\nCC:  @jdemeyer\n\nComponent: **python3**\n\nAuthor: **Fr\u00e9d\u00e9ric Chapoton**\n\nBranch/Commit: **[`42ef83b`](https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b)**\n\nReviewer: **Jeroen Demeyer**\n\n_Issue created by migration from https://trac.sagemath.org/ticket/22335_\n\n",
    "closed_at": "2017-02-23T12:00:36Z",
    "created_at": "2017-02-09T12:36:39Z",
    "labels": [
        "https://github.com/sagemath/sage/labels/c%3A%20python3",
        "https://github.com/sagemath/sage/labels/p%3A%20major%20/%203",
        "https://github.com/sagemath/sage/labels/enhancement"
    ],
    "milestone": "https://github.com/sagemath/sage/milestones/sage-7.6",
    "reactions": [],
    "repository": "https://github.com/sagemath/sage",
    "title": "PyInt_AS_LONG is gone in Python 3",
    "type": "issue",
    "updated_at": "2017-02-23T12:00:36Z",
    "url": "https://github.com/sagemath/sage/issues/22335",
    "user": "https://github.com/fchapoton"
}
```
<div id="comment:0"></div>

The C++ code in `triangulations.cc` (which does not involve Cython) calls `PyInt_AS_LONG` which no longer exists in Python 3:

```
[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc: In function 'triangulations* init_triangulations(int, int, int, bool, PyObject*, PyObject*)':
[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc:121:41: error: 'PyInt_AS_LONG' was not declared in this scope
```

Luckily, Cython defines `PyInt_AS_LONG` as an alias of `PyLong_AS_LONG` in Python 3. So there should be no compilation issues for Cython code.

I include a branch to allow others to reproduce:

```
export SAGE_PYTHON3=yes
make build
```

CC:  @jdemeyer

Component: **python3**

Author: **Frédéric Chapoton**

Branch/Commit: **[`42ef83b`](https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b)**

Reviewer: **Jeroen Demeyer**

_Issue created by migration from https://trac.sagemath.org/ticket/22335_





---

archive/issue_events_309815.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2017-02-09T12:36:39Z",
    "event": "milestoned",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "milestone_number": null,
    "milestone_title": "sage-7.6",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309815"
}
```



---

archive/issue_events_309816.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2017-02-09T12:36:39Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/p%3A%20major%20/%203",
    "label_color": "ffbb00",
    "label_name": "p: major / 3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309816"
}
```



---

archive/issue_events_309817.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2017-02-09T12:36:56Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/c%3A%20python3",
    "label_color": "0000b0",
    "label_name": "c: python3",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309817"
}
```



---

archive/issue_comments_330123.json:
```json
{
    "body": "<div id=\"comment:2\" align=\"right\">comment:2</div>\n\nOne problem is that `PyInt_AS_LONG` is often used in performance-critical code.\n\nSo whatever you do **be aware of performance regressions** in this ticket. Although it could very well be that such regressions are inevitable.",
    "created_at": "2017-02-09T20:12:06Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330123",
    "user": "https://github.com/jdemeyer"
}
```

<div id="comment:2" align="right">comment:2</div>

One problem is that `PyInt_AS_LONG` is often used in performance-critical code.

So whatever you do **be aware of performance regressions** in this ticket. Although it could very well be that such regressions are inevitable.



---

archive/issue_comments_330124.json:
```json
{
    "body": "Description changed:\n``````diff\n--- \n+++ \n@@ -1,9 +1,5 @@\n-once the last calls to cmp() in pyx files will be removed, the next failure will be this one:\n+There are 43 places in Sage where `PyInt_AS_LONG' is used, but that function no longer exists in Python 3.\n \n-```\n-[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc: In function 'triangulations* init_triangulations(int, int, int, bool, PyObject*, PyObject*)':\n-[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc:121:41: error: 'PyInt_AS_LONG' was not declared in this scope\n-```\n I include a branch to allow others to reproduce:\n \n ```\n``````\n",
    "created_at": "2017-02-09T20:14:20Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330124",
    "user": "https://github.com/jdemeyer"
}
```

Description changed:
``````diff
--- 
+++ 
@@ -1,9 +1,5 @@
-once the last calls to cmp() in pyx files will be removed, the next failure will be this one:
+There are 43 places in Sage where `PyInt_AS_LONG' is used, but that function no longer exists in Python 3.
 
-```
-[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc: In function 'triangulations* init_triangulations(int, int, int, bool, PyObject*, PyObject*)':
-[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc:121:41: error: 'PyInt_AS_LONG' was not declared in this scope
-```
 I include a branch to allow others to reproduce:
 
 ```
``````




---

archive/issue_events_309818.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2017-02-09T20:14:20Z",
    "event": "renamed",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "title_is": "PyInt_AS_LONG is gone in Python 3",
    "title_was": "py3 : compilation problem involving PyInt_AS_LONG",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309818"
}
```



---

archive/issue_comments_330125.json:
```json
{
    "body": "Description changed:\n``````diff\n--- \n+++ \n@@ -1,4 +1,11 @@\n-There are 43 places in Sage where `PyInt_AS_LONG' is used, but that function no longer exists in Python 3.\n+The C++ code in `triangulations.cc` (which does not involve Cython) calls `PyInt_AS_LONG` which no longer exists in Python 3:\n+\n+```\n+[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc: In function 'triangulations* init_triangulations(int, int, int, bool, PyObject*, PyObject*)':\n+[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc:121:41: error: 'PyInt_AS_LONG' was not declared in this scope\n+```\n+\n+Luckily, Cython defines `PyInt_AS_LONG` as an alias of `PyLong_AS_LONG` in Python 3. So there should be no compilation issues for Cython code.\n \n I include a branch to allow others to reproduce:\n \n``````\n",
    "created_at": "2017-02-09T20:26:33Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330125",
    "user": "https://github.com/jdemeyer"
}
```

Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,11 @@
-There are 43 places in Sage where `PyInt_AS_LONG' is used, but that function no longer exists in Python 3.
+The C++ code in `triangulations.cc` (which does not involve Cython) calls `PyInt_AS_LONG` which no longer exists in Python 3:
+
+```
+[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc: In function 'triangulations* init_triangulations(int, int, int, bool, PyObject*, PyObject*)':
+[sagelib-7.6.beta3] sage/geometry/triangulation/triangulations.cc:121:41: error: 'PyInt_AS_LONG' was not declared in this scope
+```
+
+Luckily, Cython defines `PyInt_AS_LONG` as an alias of `PyLong_AS_LONG` in Python 3. So there should be no compilation issues for Cython code.
 
 I include a branch to allow others to reproduce:
 
``````




---

archive/issue_comments_330126.json:
```json
{
    "body": "<div id=\"comment:4\" align=\"right\">comment:4</div>\n\nIt's actually not as bad as I thought since Cython does support `PyInt_AS_LONG`.",
    "created_at": "2017-02-09T20:26:33Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330126",
    "user": "https://github.com/jdemeyer"
}
```

<div id="comment:4" align="right">comment:4</div>

It's actually not as bad as I thought since Cython does support `PyInt_AS_LONG`.



---

archive/issue_events_309819.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2017-02-09T20:26:33Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/enhancement",
    "label_color": "696969",
    "label_name": "enhancement",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309819"
}
```



---

archive/issue_comments_330127.json:
```json
{
    "body": "<div id=\"comment:5\" align=\"right\">comment:5</div>\n\nAs the author I'd say replacing it with PyLong_AsLong should fix it.",
    "created_at": "2017-02-09T21:07:17Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330127",
    "user": "https://github.com/vbraun"
}
```

<div id="comment:5" align="right">comment:5</div>

As the author I'd say replacing it with PyLong_AsLong should fix it.



---

archive/issue_comments_330128.json:
```json
{
    "body": "<div id=\"comment:6\" align=\"right\">comment:6</div>\n\nI think I do not understand the conclusion..\n\nWell, as this is way above my head, could please one of you handle that ?",
    "created_at": "2017-02-09T21:10:55Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330128",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:6" align="right">comment:6</div>

I think I do not understand the conclusion..

Well, as this is way above my head, could please one of you handle that ?



---

archive/issue_comments_330129.json:
```json
{
    "body": "<div id=\"comment:7\" align=\"right\">comment:7</div>\n\nAlternatively, we could `#include` those C++ files in the Cython sources instead of compiling them separately. Then `PyInt_AS_LONG` would be handled by Cython.",
    "created_at": "2017-02-10T07:41:01Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330129",
    "user": "https://github.com/jdemeyer"
}
```

<div id="comment:7" align="right">comment:7</div>

Alternatively, we could `#include` those C++ files in the Cython sources instead of compiling them separately. Then `PyInt_AS_LONG` would be handled by Cython.



---

archive/issue_comments_330130.json:
```json
{
    "body": "<div id=\"comment:8\" align=\"right\">comment:8</div>\n\nI tried by replacing with PyLong_AsLong. It seems to work (in python3).\n\nNext problem (similar ?):\n\n```\n[sagelib-7.6.beta3] /home/chapoton/sage3/src/sage/libs/pynac/wrap.h: In function 'PyObject* _to_PyString_latex(const T*)':\n[sagelib-7.6.beta3] /home/chapoton/sage3/src/sage/libs/pynac/wrap.h:118:50: error: there are no arguments to 'PyString_FromString' that depend on a template parameter, so a declaration of 'PyString_FromString' must be available [-fpermissive]\n[sagelib-7.6.beta3]    return PyString_FromString(instore.str().data());\n[sagelib-7.6.beta3]\n[sagelib-7.6.beta3] /home/chapoton/sage3/src/build/cythonized/sage/libs/pynac/pynac.cpp:5478:47: error: 'PyString_AsString' was not declared in this scope\n```\n\nEDIT: see #22341",
    "created_at": "2017-02-10T08:59:45Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330130",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:8" align="right">comment:8</div>

I tried by replacing with PyLong_AsLong. It seems to work (in python3).

Next problem (similar ?):

```
[sagelib-7.6.beta3] /home/chapoton/sage3/src/sage/libs/pynac/wrap.h: In function 'PyObject* _to_PyString_latex(const T*)':
[sagelib-7.6.beta3] /home/chapoton/sage3/src/sage/libs/pynac/wrap.h:118:50: error: there are no arguments to 'PyString_FromString' that depend on a template parameter, so a declaration of 'PyString_FromString' must be available [-fpermissive]
[sagelib-7.6.beta3]    return PyString_FromString(instore.str().data());
[sagelib-7.6.beta3]
[sagelib-7.6.beta3] /home/chapoton/sage3/src/build/cythonized/sage/libs/pynac/pynac.cpp:5478:47: error: 'PyString_AsString' was not declared in this scope
```

EDIT: see #22341



---

archive/issue_comments_330131.json:
```json
{
    "body": "Changed commit from **[`2c29230`](https://github.com/sagemath/sagetrac-mirror/commit/2c2923095b5dec6fd09820d6ff09ec97c1ee8903)** to **[`8cc4478`](https://github.com/sagemath/sagetrac-mirror/commit/8cc4478dc88662473d34817a7f477ff9e287786a)**",
    "created_at": "2017-02-10T16:56:28Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330131",
    "user": "https://github.com/fchapoton"
}
```

Changed commit from **[`2c29230`](https://github.com/sagemath/sagetrac-mirror/commit/2c2923095b5dec6fd09820d6ff09ec97c1ee8903)** to **[`8cc4478`](https://github.com/sagemath/sagetrac-mirror/commit/8cc4478dc88662473d34817a7f477ff9e287786a)**



---

archive/issue_comments_330132.json:
```json
{
    "body": "<div id=\"comment:9\" align=\"right\">comment:9</div>\n\nI removed the test branch (kept as public/pyint_as_long_in_cc)\n\nand added a branch making the needed changes in triangulations.\n\n---\nNew commits:\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/8cc4478dc88662473d34817a7f477ff9e287786a\"><code>8cc4478</code></a></td><td><code>replace PyInt_* by PyLong_* in triangulations.cc</code></td></tr></table>\n",
    "created_at": "2017-02-10T16:56:28Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330132",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:9" align="right">comment:9</div>

I removed the test branch (kept as public/pyint_as_long_in_cc)

and added a branch making the needed changes in triangulations.

---
New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/8cc4478dc88662473d34817a7f477ff9e287786a"><code>8cc4478</code></a></td><td><code>replace PyInt_* by PyLong_* in triangulations.cc</code></td></tr></table>




---

archive/issue_comments_330133.json:
```json
{
    "body": "Changed branch from **[public/pyint_as_long_in_cc](https://github.com/sagemath/sagetrac-mirror/tree/public/pyint_as_long_in_cc)** to **[u/chapoton/22335](https://github.com/sagemath/sagetrac-mirror/tree/u/chapoton/22335)**",
    "created_at": "2017-02-10T16:56:28Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330133",
    "user": "https://github.com/fchapoton"
}
```

Changed branch from **[public/pyint_as_long_in_cc](https://github.com/sagemath/sagetrac-mirror/tree/public/pyint_as_long_in_cc)** to **[u/chapoton/22335](https://github.com/sagemath/sagetrac-mirror/tree/u/chapoton/22335)**



---

archive/issue_comments_330134.json:
```json
{
    "body": "<div id=\"comment:10\" align=\"right\">comment:10</div>\n\nthere is another instance in src/sage/modular/arithgroup/farey.cpp\n\nbut maybe we can see later if we need to change it also.",
    "created_at": "2017-02-10T16:58:44Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330134",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:10" align="right">comment:10</div>

there is another instance in src/sage/modular/arithgroup/farey.cpp

but maybe we can see later if we need to change it also.



---

archive/issue_events_309820.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2017-02-11T22:30:54Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/needs%20info",
    "label_color": "ffff00",
    "label_name": "needs info",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309820"
}
```



---

archive/issue_comments_330135.json:
```json
{
    "body": "<div id=\"comment:11\" align=\"right\">comment:11</div>\n\nsome failing doctest",
    "created_at": "2017-02-11T22:30:54Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330135",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:11" align="right">comment:11</div>

some failing doctest



---

archive/issue_comments_330136.json:
```json
{
    "body": "<div id=\"comment:12\" align=\"right\">comment:12</div>\n\ntypical failing doctest:\n\n```\nFile \"src/sage/geometry/triangulation/base.pyx\", line 903, in sage.geometry.triangulation.base.ConnectedTriangulationsIterator\nFailed example:\n    next(ci)\nExpected:\n    (2, 3, 4, 5)\nGot:\n    (2L, 3L, 4L, 5L)\n```\nProbably in python3, the result will stil be (2, 3, 4, 5).\n\nVolker or Jeroen, any suggestion on how to have doctests working both in py2 and py3 ?\n\nCasting these integers to Integer ? I would prefer not.\n\nOr just use apply \"str\" to display the results without the trailing L ?",
    "created_at": "2017-02-13T15:36:59Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330136",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:12" align="right">comment:12</div>

typical failing doctest:

```
File "src/sage/geometry/triangulation/base.pyx", line 903, in sage.geometry.triangulation.base.ConnectedTriangulationsIterator
Failed example:
    next(ci)
Expected:
    (2, 3, 4, 5)
Got:
    (2L, 3L, 4L, 5L)
```
Probably in python3, the result will stil be (2, 3, 4, 5).

Volker or Jeroen, any suggestion on how to have doctests working both in py2 and py3 ?

Casting these integers to Integer ? I would prefer not.

Or just use apply "str" to display the results without the trailing L ?



---

archive/issue_comments_330137.json:
```json
{
    "body": "<div id=\"comment:13\" align=\"right\">comment:13</div>\n\nWell, I don't think that replacing `PyInt...` by `PyLong...` is a correct solution anyway.\n\nI'd like to know what Volker thinks about [comment:7]",
    "created_at": "2017-02-13T15:40:14Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330137",
    "user": "https://github.com/jdemeyer"
}
```

<div id="comment:13" align="right">comment:13</div>

Well, I don't think that replacing `PyInt...` by `PyLong...` is a correct solution anyway.

I'd like to know what Volker thinks about [comment:7]



---

archive/issue_comments_330138.json:
```json
{
    "body": "<div id=\"comment:14\" align=\"right\">comment:14</div>\n\nWell I'm not really a fan of including c++ files, thats ought to be the compiler's job. All that Cython does is \n\n```\n#if PY_MAJOR_VERSION >= 3\n  #define PyInt_FromLong               PyLong_FromLong\n  #define PyInt_AsLong                 PyLong_AsLong\n  #define PyInt_AS_LONG                PyLong_AS_LONG\n#endif\n```\nI'd just put that in the offending C++ files..",
    "created_at": "2017-02-13T19:21:57Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330138",
    "user": "https://github.com/vbraun"
}
```

<div id="comment:14" align="right">comment:14</div>

Well I'm not really a fan of including c++ files, thats ought to be the compiler's job. All that Cython does is 

```
#if PY_MAJOR_VERSION >= 3
  #define PyInt_FromLong               PyLong_FromLong
  #define PyInt_AsLong                 PyLong_AsLong
  #define PyInt_AS_LONG                PyLong_AS_LONG
#endif
```
I'd just put that in the offending C++ files..



---

archive/issue_comments_330139.json:
```json
{
    "body": "Changed branch from **[u/chapoton/22335](https://github.com/sagemath/sagetrac-mirror/tree/u/chapoton/22335)** to **[public/22335](https://github.com/sagemath/sagetrac-mirror/tree/public/22335)**",
    "created_at": "2017-02-13T20:35:25Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330139",
    "user": "https://github.com/fchapoton"
}
```

Changed branch from **[u/chapoton/22335](https://github.com/sagemath/sagetrac-mirror/tree/u/chapoton/22335)** to **[public/22335](https://github.com/sagemath/sagetrac-mirror/tree/public/22335)**



---

archive/issue_comments_330140.json:
```json
{
    "body": "Changed commit from **[`8cc4478`](https://github.com/sagemath/sagetrac-mirror/commit/8cc4478dc88662473d34817a7f477ff9e287786a)** to **[`42ef83b`](https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b)**",
    "created_at": "2017-02-13T20:35:25Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330140",
    "user": "https://github.com/fchapoton"
}
```

Changed commit from **[`8cc4478`](https://github.com/sagemath/sagetrac-mirror/commit/8cc4478dc88662473d34817a7f477ff9e287786a)** to **[`42ef83b`](https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b)**



---

archive/issue_comments_330141.json:
```json
{
    "body": "<div id=\"comment:15\" align=\"right\">comment:15</div>\n\nok, I have tried to add the suggested lines.\nI put them in the .h files, near the top.\nPlease tell me if this is ok.\nI am going to launch my bot on it now.\n\n---\nNew commits:\n<table><tr><td><a href=\"https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b\"><code>42ef83b</code></a></td><td><code>py3, trac 22335 adding alias for PyInt in C files for use with python3</code></td></tr></table>\n",
    "created_at": "2017-02-13T20:35:25Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330141",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:15" align="right">comment:15</div>

ok, I have tried to add the suggested lines.
I put them in the .h files, near the top.
Please tell me if this is ok.
I am going to launch my bot on it now.

---
New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b"><code>42ef83b</code></a></td><td><code>py3, trac 22335 adding alias for PyInt in C files for use with python3</code></td></tr></table>




---

archive/issue_comments_330142.json:
```json
{
    "body": "<div id=\"comment:16\" align=\"right\">comment:16</div>\n\nbot seems to be green, please review",
    "created_at": "2017-02-14T08:12:16Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330142",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:16" align="right">comment:16</div>

bot seems to be green, please review



---

archive/issue_events_309821.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2017-02-14T10:10:35Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/needs%20info",
    "label_color": "ffff00",
    "label_name": "needs info",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309821"
}
```



---

archive/issue_events_309822.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2017-02-14T10:10:35Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309822"
}
```



---

archive/issue_comments_330143.json:
```json
{
    "body": "<div id=\"comment:18\" align=\"right\">comment:18</div>\n\nping ?",
    "created_at": "2017-02-17T18:21:02Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330143",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:18" align="right">comment:18</div>

ping ?



---

archive/issue_comments_330144.json:
```json
{
    "body": "Reviewer: **Jeroen Demeyer**",
    "created_at": "2017-02-17T18:59:06Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330144",
    "user": "https://github.com/jdemeyer"
}
```

Reviewer: **Jeroen Demeyer**



---

archive/issue_comments_330145.json:
```json
{
    "body": "<div id=\"comment:19\" align=\"right\">comment:19</div>\n\nPositive review but author name is missing...",
    "created_at": "2017-02-17T18:59:06Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330145",
    "user": "https://github.com/jdemeyer"
}
```

<div id="comment:19" align="right">comment:19</div>

Positive review but author name is missing...



---

archive/issue_events_309823.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2017-02-17T18:59:06Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/needs%20review",
    "label_color": "7fff00",
    "label_name": "needs review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309823"
}
```



---

archive/issue_events_309824.json:
```json
{
    "actor": "https://github.com/jdemeyer",
    "created_at": "2017-02-17T18:59:06Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309824"
}
```



---

archive/issue_comments_330146.json:
```json
{
    "body": "Author: **Fr\u00e9d\u00e9ric Chapoton**",
    "created_at": "2017-02-17T19:32:17Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330146",
    "user": "https://github.com/fchapoton"
}
```

Author: **Frédéric Chapoton**



---

archive/issue_comments_330147.json:
```json
{
    "body": "<div id=\"comment:20\" align=\"right\">comment:20</div>\n\nthanks",
    "created_at": "2017-02-17T19:32:17Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330147",
    "user": "https://github.com/fchapoton"
}
```

<div id="comment:20" align="right">comment:20</div>

thanks



---

archive/issue_events_309825.json:
```json
{
    "actor": "https://github.com/vbraun",
    "created_at": "2017-02-23T12:00:36Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "label": "https://github.com/sagemath/sage/labels/positive%20review",
    "label_color": "dfffc0",
    "label_name": "positive review",
    "label_text_color": "ffffff",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309825"
}
```



---

archive/issue_events_309826.json:
```json
{
    "actor": "https://github.com/vbraun",
    "commit_id": "c42ed0ea94f2fc0b0ef0c91d5f901a394e70f06a",
    "commit_repository": "https://github.com/sagemath/sage",
    "created_at": "2017-02-23T12:00:36Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/22335#event-309826"
}
```



---

archive/issue_comments_330148.json:
```json
{
    "body": "Changed branch from **[public/22335](https://github.com/sagemath/sagetrac-mirror/tree/public/22335)** to **[`42ef83b`](https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b)**",
    "created_at": "2017-02-23T12:00:36Z",
    "formatter": "markdown",
    "issue": "https://github.com/sagemath/sage/issues/22335",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/22335#issuecomment-330148",
    "user": "https://github.com/vbraun"
}
```

Changed branch from **[public/22335](https://github.com/sagemath/sagetrac-mirror/tree/public/22335)** to **[`42ef83b`](https://github.com/sagemath/sagetrac-mirror/commit/42ef83b01aaef18f50674a82098e095566e24c1b)**
